tests/package/generic/002.phpt

```diff
@@ -4,30 +4,14 @@ CALS Table rendering#002
 <?php
 namespace phpdotnet\phd;
 
-require_once __DIR__ . "/../setup.php";
-require_once __DIR__ . "/TestChunkedXHTML.php";
+require_once __DIR__ . "/../../setup.php";
 
-$formatclass = "TestChunkedXHTML";
 $xml_file = __DIR__ . "/data/002.xml";
 
-$opts = array(
-    "index"             => true,
-    "xml_root"          => dirname($xml_file),
-    "xml_file"          => $xml_file,
-    "output_dir"        => __DIR__ . "/output/",
-);
+Config::init(["xml_file" => $xml_file]);
```

Comment:
Do you mean something like a `readonly` Value Object with named constructor parameters? Because that sounds like something that might make juggling these configurations easier. Unfortunately the properties of `Config` are not setup in the beginning of rendering so that it can easily be changed to a `readonly` VO but with some refactoring that could be done too. 

I'll make a note of this and will probably start refactoring parts of this one by one (such as `Options_Parser` to return an array instead of setting `Config` directly, `Index`'s database to be constructor injected so it can be added to `Config` outside of `Index` as well, moving hardcoded `css` urls out of PHP's XHTML format so it also can be added to `Config` at the very beginning of the script, `output_format`s also set at the beginning of the script, etc.).

Reply:
Basically yes :) having a magic dictionary array with keys is hard to reason about, could be fixed by using a static analyser array shape, but I doubt any static analysis tool is going to enjoy PhD in the state it currently is :')